File: test/corpus/java/flags.java
// Presence flags: a stub marker, a swallowed exception, and a return inside a
// finally. Three units, under the cohesion floor.

class FlagOps {
    static final int RETRIES = 3;

    String loadConfig(String path) {
        // dendro-expect: stub_marker
        // TODO: validate the parsed config before returning
        return readFile(path);
    }

    int fetch(String url) {
        for (int i = 0; i < RETRIES; i++) {
            try {
                return request(url);
                // dendro-expect: empty_catch
            } catch (Exception e) {
            }
        }
        return 0;
    }

    void closeQuietly(Handle handle) {
        try {
            handle.flush();
        } finally {
            // dendro-expect: return_in_finally
            return;
        }
    }
}

File: test/corpus/python/flags.py
# Presence flags: a stub marker, a swallowed exception, and a return inside a
# finally. Three units, under the cohesion floor.

RETRIES = 3


def load_config(path):
    # dendro-expect: stub_marker
    # TODO: validate the parsed config before returning
    with open(path) as fh:
        return fh.read()


def fetch(url):
    for _ in range(RETRIES):
        try:
            return request(url)
        # dendro-expect: empty_catch
        except Exception:
            pass
    return None


def close_quietly(handle):
    try:
        handle.flush()
    finally:
        # dendro-expect: return_in_finally
        return
